module/zfs/vdev_raidz.c

```diff
@@ -722,6 +678,63 @@ vdev_raidz_generate_parity_pqr(raidz_map_t *rm)
 	}
 }
 
+#if defined(__x86_64__)
+void vdev_raidz_generate_parity_p_avx128(raidz_map_t *rm);
+void vdev_raidz_generate_parity_pq_avx128(raidz_map_t *rm);
+void vdev_raidz_generate_parity_pqr_avx128(raidz_map_t *rm);
+void vdev_raidz_generate_parity_p_avx2(raidz_map_t *rm);
+void vdev_raidz_generate_parity_pq_avx2(raidz_map_t *rm);
+void vdev_raidz_generate_parity_pqr_avx2(raidz_map_t *rm);
+void vdev_raidz_generate_parity_p_sse(raidz_map_t *rm);
+void vdev_raidz_generate_parity_pq_sse(raidz_map_t *rm);
+void vdev_raidz_generate_parity_pqr_sse(raidz_map_t *rm);
+#endif
+
+static void vdev_raidz_pick_parity_functions(void) {
+	vdev_raidz_generate_parity_p = &vdev_raidz_generate_parity_p_c;
+	vdev_raidz_generate_parity_pq = &vdev_raidz_generate_parity_pq_c;
+	vdev_raidz_generate_parity_pqr = &vdev_raidz_generate_parity_pqr_c;
```

Comment:
Given that these 3 are tied together, it would be easier to read if you declared a struct with these 3 function pointers, and in this func, just assign a global pointer to the appropriate struct (see zfs_metaslab_ops for an example).